.github/workflows/check-pr-engine-editor-debug-and-tests.yaml

```diff
@@ -0,0 +1,90 @@
+name: Check PR - Engine Editor Debug And Tests
+on:
+  pull_request:
+    paths:
+      - 'src/**'
+      - 'config.py'
+      - 'register_types.cpp'
+      - 'register_types.h'
+      - 'SCsub'
+      - 'kt/**'
+      - 'harness/tests/**'
+      - '.github/workflows/check-pr-engine-editor-debug-and-tests.yaml'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        include:
+          - os: ubuntu-latest
+            platform: x11
+          - os: macos-latest
+            platform: osx
+          - os: windows-latest
+            platform: windows
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Configure dependencies
+        if: matrix.platform == 'x11'
+        run: |
+          sudo apt-get install build-essential pkg-config libx11-dev libxcursor-dev \
+            libxinerama-dev libgl1-mesa-dev libglu-dev libasound2-dev libpulse-dev libudev-dev libxi-dev libxrandr-dev yasm
+      - name: Clone Godot Engine
+        uses: actions/checkout@v2
+        with:
+          repository: godotengine/godot
+          ref: 3.2.3-stable
+      - name: Clone Godot JVM module.
+        uses: actions/checkout@v2
+        with:
+          path: modules/kotlin_jvm
+          submodules: recursive
+      - name: Set up Python 3.x
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+          architecture: 'x64'
+      - name: Configuring Python packages
+        run: |
+          python -c "import sys; print(sys.version)"
+          python -m pip install scons
+      - name: Build with editor debug
+        run: |
+          scons platform=${{ matrix.platform }}
+      - name: Upload OSX binary
+        if: matrix.platform == 'osx'
+        uses: actions/upload-artifact@v1
+        with:
+          name: osx-editor-debug-app
+          path: bin/godot.osx.tools.64
+  test:
+    needs: build
+    runs-on: macos-latest
+    steps:
+      - name: Clone Godot JVM module.
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      - name: Build tests project
+        uses: eskatos/gradle-command-action@v1
+        with:
+          wrapper-directory: harness/tests/
+          build-root-directory: harness/tests/
+          arguments: build
+      - name: Download OSX Editor Debug App
+        uses: actions/download-artifact@v1
+        with:
+          name: osx-editor-debug-app
+      - name: Run Tests
+        run: |
+          cd harness/tests/
+          wget https://gist.githubusercontent.com/piiertho/fb25f9326189d00608ffeb7702b91395/raw/dddef0b20a61062bb4d091bf2f7ada6b7b25b7c7/run_godot_kotlin_tests.sh

```

Comment:
Why a gist and not a script inside the repo?

Reply:
Because I plan to use it in other projects :p